.ibm/pipelines/utils.sh

```diff
@@ -285,3 +287,441 @@ install_pipelines_operator() {
+
+    escaped_url=$(printf '%s\n' "${ENCODED_API_SERVER_URL}" | sed 's/[\/&]/\\&/g')
+    sed -i "s/K8S_CLUSTER_API_SERVER_URL:.*/K8S_CLUSTER_API_SERVER_URL: ${escaped_url}/g" "$dir/auth/secrets-rhdh-secrets.yaml" \
+      && echo "Updated K8S_CLUSTER_API_SERVER_URL in secrets file." \
+      || echo "Failed to update K8S_CLUSTER_API_SERVER_URL." >&2
+
+    sed -i "s/K8S_CLUSTER_NAME:.*/K8S_CLUSTER_NAME: ${ENCODED_CLUSTER_NAME}/g" "$dir/auth/secrets-rhdh-secrets.yaml"
+
+    token=$(oc get secret "${secret_name}" -n "${project}" -o=jsonpath='{.data.token}')
+    sed -i "s/OCM_CLUSTER_TOKEN: .*/OCM_CLUSTER_TOKEN: ${token}/" "$dir/auth/secrets-rhdh-secrets.yaml"
+
+    # Select the configuration file based on the namespace or job
+    config_file=$(select_config_map_file)
+    # Apply the ConfigMap with the correct file
+    create_app_config_map "$config_file" "$project"
+    oc create configmap rbac-policy \
+      --from-file="rbac-policy.csv"="$dir/resources/config_map/rbac-policy.csv" \
+      --namespace="$project" \
+      --dry-run=client -o yaml | oc apply -f -
+
+    oc apply -f "$dir/auth/secrets-rhdh-secrets.yaml" --namespace="${project}"
+
+}
+
+deploy_test_backstage_provider() {
+  local project=$1
+  echo "Deploying test-backstage-customization-provider in namespace ${project}"
+
+  # Check if the buildconfig already exists
+  if ! oc get buildconfig test-backstage-customization-provider -n "${project}" >/dev/null 2>&1; then
+    echo "Creating new app for test-backstage-customization-provider"
+    oc new-app https://github.com/janus-qe/test-backstage-customization-provider --namespace="${project}"
+  else
+    echo "BuildConfig for test-backstage-customization-provider already exists in ${project}. Skipping new-app creation."
+  fi
+
+  # Ensure the service exists
+  if ! oc get service test-backstage-customization-provider -n "${project}" >/dev/null 2>&1; then
+    echo "Exposing service for test-backstage-customization-provider"
+    oc expose svc/test-backstage-customization-provider --namespace="${project}"
+  else
+    echo "Service test-backstage-customization-provider is already exposed in ${project}."
+  fi
+}
+
+create_app_config_map() {
+  local config_file=$1
+  local project=$2
+
+  oc create configmap app-config-rhdh \
+    --from-file="app-config-rhdh.yaml"="$config_file" \
+    --namespace="$project" \
+    --dry-run=client -o yaml | oc apply -f -
+}
+
+select_config_map_file() {
+  if [[ "${project}" == *rbac* ]]; then
+    echo "$dir/resources/config_map/app-config-rhdh-rbac.yaml"
+  else
+    echo "$dir/resources/config_map/app-config-rhdh.yaml"
+  fi
+}
+
+create_app_configmap_aks_gke() {
+    local dir=$1
+    local project=$2
+
+    echo "Creating app-config ConfigMap for AKS/GKE in namespace ${project}"
+
+    # Modifica o YAML e aplica no cluster
+    yq 'del(.backend.cache)' "${dir}/resources/config_map/app-config-rhdh.yaml" \
```

Comment:
To use it with `select_config_map_file`, we need to modify it.
```suggestion
    yq 'del(.backend.cache)' "$config_file" \
```